docs/specs/clients/notify/data-structures.md

```diff
@@ -56,4 +56,21 @@
   "lastModified": Int64,
   "types": NotifyType[]
 }
-```
\ No newline at end of file
+```
+
+## Notify Server Subscriptions
+
+```jsonc
+NotifyServerSubscription[]
+```
+
+`NotifyServerSubscription`:
+```typescript
+{
+  topic: string, // dApp subscription topic to manage the subscription and call wc_notifySubscriptionUpdate and wc_notifySubscriptionDelete
+  account: Account, // CAIP-10 account
+  scope: string[], // Array of notification types enabled for this subscription
+  symKey: string, // Symetric key used for notification topic. sha256 to get notify topic
+  expiry: number, // Unix timestamp of expiration
+}
```

Comment:
`metadata` is missing here, is this intentional?

Reply:
What is in metadata?

Reply:
exactly and scope doesn't match what i have:

```swift
public struct NotifySubscription: DatabaseObject {
    public let topic: String
    public let account: Account
    public let relay: RelayProtocolOptions
    public let metadata: AppMetadata
    public let scope: [String: ScopeValue]
    public let expiry: Date
    public let symKey: String

}

public struct RelayProtocolOptions: Codable, Equatable {
    public let `protocol`: String
    public let data: String?
}

public struct ScopeValue: Codable, Equatable {
    let description: String
    let enabled: Bool
}

```

Reply:
Isn't all this data supposed to be retrieved from either [`/.well-known/wc-notify-config.json`](https://specs.walletconnect.com/2.0/specs/clients/notify/notify-notification-types) or the explorer listing? That info should be refreshed occasionally client-side and doesn't need to be stored in the notify server.

Reply:
make sense, we can do it this way for scopes.
How about metadata, the sdk does not depend on the explorer and I wouldn't like to.
if notify does not need metadata, we could add it to [/.well-known/wc-notify-config.json](https://specs.walletconnect.com/2.0/specs/clients/notify/notify-notification-types) too

Reply:
> How about metadata, the sdk does not depend on the explorer and I wouldn't like to.
if notify does not need metadata, we could add it to [/.well-known/wc-notify-config.json](https://specs.walletconnect.com/2.0/specs/clients/notify/notify-notification-types) too

Not suggesting to use explorer, just wondering how you populated `metadata` before? Notify doesn't need this information, I assumed clients already had it from before.

Reply:
client have it from public method call.
so basically w3i depends on explorer and w3i has been passing metadata to notifyClient.

```swift
public func subscribe(metadata: AppMetadata, account: Account, onSign: @escaping SigningCallback) async throws {
}
```

Reply:
I think it's not a crime to add metadata to config.json

Reply:
I see. I think it makes sense for the wallet client to handle this, question is if we add it to wc-notify-config.json or make a callback to W3I SDK?

I think it should be updated periodically in-case dapp changes their e.g. descriptions or icon, rather than being permanently stored in what was previously sync API.

Reply:
> or make a callback to W3I SDK?

lower lever module(notify) should not depend on higher level module(w3i)
also we are allowing to integrate notify natively without w3i.

>I think it should be updated periodically

storing metadata in wc-notify-config.json and periodically resolving it would do this job
currently we are cashing it in run time memory

Reply:
but there is one more problem, how can we resolve dappUrl from?:

```
class NotifyServerSubscription: Codable, Equatable {
    let topic: String
    let account: Account
    let scope: [String]
    let symKey: String
    let expiry: Date
}
```

Reply:
I know I have a subscription but I have no idea what dapp does the subscription refers to...

Reply:
We can add dappUrl, notify server already has that from /subscribe-topic

Reply:
perfect, so please add

Reply:
but what do we do about the metadata? :)

Reply:
> but what do we do about the metadata? :)

Sounds like we could do this:
> if notify does not need metadata, we could add it to [/.well-known/wc-notify-config.json](https://specs.walletconnect.com/2.0/specs/clients/notify/notify-notification-types) too

Personally I don't see a problem with this:
> > or make a callback to W3I SDK?
> 
> lower lever module(notify) should not depend on higher level module(w3i)
also we are allowing to integrate notify natively without w3i.

The purpose of the callback is to decouple the notify SDK from the W3I SDK. Instead of metadata being passed to notify SDK, a callback that returns metadata would be passed. So it can be implemented by W3I SDK to pull from explorer, or someing integrating notify SDK directly can implement the method to return hardcoded values.

Reply:
>The purpose of the callback is to decouple the notify SDK from the W3I SDK. Instead of metadata being passed to notify SDK, a callback that returns metadata would be passed. So it can be implemented by W3I SDK to pull from explorer, or someing integrating notify SDK directly can implement the method to return hardcoded values.

yeah but this would require wallets to use explorer making the whole integration process much more time consuming

let's add it to config file unless you are inclined towards other solution
@devceline @Elyniss  what do you think?

Reply:
Adding it to the config file sounds perfect, assuming we also add dappUrl

Reply:
I'm in-favor of using wc-notify-config.json because issue with callback is wallet would need to definitely depend on explorer then.